galsim/scene.py

```diff
@@ -0,0 +1,412 @@
+#
+"""@file scene.py
+Routines for defining a "sky scene", i.e., a galaxy or star sample with reasonable properties that
+can then be placed throughout a large image.  Currently, this only includes routines for making a
+COSMOS-based galaxy sample, but it could be expanded to include star samples as well.
+"""
+
+import galsim
+from galsim import pyfits
+import numpy as np
+import math
+import os
+
+# Below is a number that is needed to relate the COSMOS parametric galaxy fits to quantities that
+# GalSim needs to make a GSObject representing that fit.  It is simply the pixel scale, in arcsec,
+# in the COSMOS weak lensing reductions used for the fits.
+cosmos_pix_scale = 0.03
+
+class COSMOSCatalog(object):
+    """
+    A class representing a sample of galaxies from the COSMOS sample with F814W<23.5.
+
+    Depending on the keyword arguments, particularly `use_real`, the catalog will either have
+    information about real galaxies, or parametric ones.  To use this with either type of galaxies,
+    you need to get the COSMOS datasets in the format that GalSim recognizes; see
+
+        https://github.com/GalSim-developers/GalSim/wiki/RealGalaxy-Data
+
+    option (1) for more information.  Note that if you want to make real galaxies you need to
+    download and store the full tarball with all galaxy images, whereas if you want to make
+    parametric galaxies you only need the catalog real_galaxy_catalog_23.5_fits.fits and can delete
+    the other files.
+
+    Finally, we provide a program that will download the large COSMOS sample for you and
+    put it in the $PREFIX/share/galsim directory of your installation path.  The program is
+
+            galsim_download_cosmos
+
+    which gets installed in the $PREFIX/bin directory when you install GalSim.  If you use
+    this program to download the COSMOS catalog, then you can use it with
+
+            cat = galsim.COSMOSCatalog()
+
+    GalSim knows the location of the installation share directory, so it will automatically
+    look for it there.
+
+    After getting the catalogs, there is a method makeObj() that can make an object corresponding to
+    any chosen galaxy in the catalog (whether real or parametric).  See
+    help(galsim.COSMOSCatalog.makeObj) for more information.  As an interesting application and
+    example of the usage of these routines, consider the following code:
+
+        >>>> im_size = 64
+        >>>> pix_scale = 0.05
+        >>>> bandpass = galsim.Bandpass('share/wfc_F814W.dat.gz',
+                                        wave_type='ang').thin().withZeropoint(25.94)
+        >>>> real_cat = galsim.COSMOSCatalog('real_galaxy_catalog_23.5.fits',
+                                             dir='/path/to/COSMOS/data')
+        >>>> param_cat = galsim.COSMOSCatalog('real_galaxy_catalog_23.5_fits.fits',
+                                              dir='/path/to/COSMOS/data',
+                                              use_real=False, exclude_fail=False)
```

Comment:
I think now these can just be
```
>>> real_cat = galsim.COSMOSCatalog()
>>> param_cat = galsim.COSMOSCatalog(use_real=False)  
```
I'm not sure we ought to suggest `exclude_fail=False` here.  Isn't it normally a better idea to not do that?  So I think it might confuse people to see this in the example.

Also, very minor point, but we usually use only 3 > symbols.  `>>>`, not `>>>>`.